gvfs-helper.c

```diff
@@ -1049,7 +1049,9 @@ static void create_tempfile_for_loose(
 	strbuf_complete(&buf_path, '/');
 	strbuf_add(&buf_path, hex, 2);
 
-	if (!file_exists(buf_path.buf) && mkdir(buf_path.buf, 0777) == -1) {
+	if (!file_exists(buf_path.buf) &&
```

Comment:
A bit unrelated, do you know if there is a function `dir_exists` that checks if the path exists as a directory (or should we add one to use here)?

I couldn't find one in `dir.c`, but I imagine all it would need is:

```
int dir_exists(const char *f)
{
	struct stat sb;
	if (lstat(f, &sb) != 0)
		return false;

	return S_ISDIR(sb.st_mode);
}
```

Reply:
The following exists in `builtin/clone.c`:

```
static int dir_exists(const char *path)
{
	struct stat sb;
	return !stat(path, &sb);
}
```

It doesn't check the `S_ISDIR()`.

I do see your interest in checking the type to make sure there isn't a file in the way. I'm not sure if there is enough value.

Reply:
> I'm not sure if there is enough value.

Yeah I was on the fence as well, we've not seen this be a problem thus far.

Reply:
Good catch.  Thanks!
I'll fix it Monday.  I think I can just try the mkdir() and catch the EEXIST error code.
I avoided that because of the platform nature of the error codes.
(or just try the lstat() again inside the error case.)

Reply:
Please note that there is not only a `dir_exists()`, but also an `is_directory()` function (the latter of which _is_ public). There is even a ticket to reconcile them: https://github.com/gitgitgadget/git/issues/230